packages/as-types/assembly/serialization/numbers.ts

```diff
@@ -23,9 +33,13 @@ export function byteToU8(arr: StaticArray<u8>): u8 {
 }
 
 /**
- * Convert a number to StaticArray<u8>
+ * Convert an integer number to StaticArray<u8>
+ *
+ * @typeParam T - the type of the number to convert (it must be an integer type)
+ *
+ * @param val - the number to convert
  *
- *  @param val - the number to convert
+ * @returns the converted StaticArray<u8>
  */
 export function toBytes<T>(val: T): StaticArray<u8> {
   if (!isInteger<T>()) {
```

Comment:
it would be good to warn the reader of what happens if "isInteger<T>()" is false